File: CPP/Vehicle.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Vehicle {
protected:
    string plateNumber;
    string brand;
    int year;
    string color;
public:
    Vehicle(string plateNumber, string brand, int year, string color)
        : plateNumber(plateNumber), brand(brand), year(year), color(color) {}

    virtual void displayInfo() {
        cout << "Plate Number: " << plateNumber << endl;
        cout << "Brand: " << brand << endl;
        cout << "Year: " << year << endl;
        cout << "Color: " << color << endl;
    }
};

File: Vehicle.java
package Java;

public class Vehicle {
    protected String plateNumber;
    protected String brand;
    protected int year;
    protected String color;

    public Vehicle(String plateNumber, String brand, int year, String color) {
        this.plateNumber = plateNumber;
        this.brand = brand;
        this.year = year;
        this.color = color;
    }

    public void displayInfo() {
        System.out.println("Plate Number: " + plateNumber);
        System.out.println("Brand: " + brand);
        System.out.println("Year: " + year);
        System.out.println("Color: " + color);
    }
}

